services/publisher_r2.py

```
@@ -170,8 +224,141 @@
                 logger.info(f"✓ Uploaded {r2_key}")
                 results[str(relative_path)] = True
                 
+                # Purge cache for this blog post
+                blog_date = blog_data.get('date')
+                if blog_date:
+                    self.cache_manager.purge_blog_cache(blog_date, self.api_domain, self.frontend_domain)
+                
             except Exception as e:
```

Ruff (0.12.2)

Do not catch blind exception: `Exception`

(BLE001)